### ISSUE TITLE
FIM System tests: Improvements for alerts verification scripts

### ISSUE THREAD

**Issue by rshad** [wazuh/wazuh-qa#544]
Hi all!

The alerts verification scripts are improvable and especially talking about the script 
- `wazuh-qa/tests/system/fim/common_tasks/verify_alerts_elasticsearch.py`

**Desired Changes List**
- [x] More efficient and Cleaner Code.
- [x] Functional Code Style: Encapsulate code in functions if it's possible.
- [x] Improve `Retry` control.

Always verifying the changes with the different scenarios listed in #493 .

Kr,

Rshad

**Comment by rshad** [comment]
Working branch: [feature-544-scripts-improvements](https://github.com/wazuh/wazuh-qa/tree/feature-544-scripts-improvements).

**Comment by rshad** [comment]
Hi all!

We are practically done with all the required changes and also added more improvements that came up during the development process.

- **Functional Programming**

We decomposed the initial version of the script `wazuh-qa/tests/system/fim/common_tasks/verify_alerts_elasticsearch.py`  into different functions with the purpose to make our code functional and easier to control.

- - **`Function: verify_es_alerts`**

This function represents the central function which practically represents the controller function by which we call the corresponding functions depending on the selected scenario and the provided arguments.

We can distinguish 3 different cases,
- `general`: Represents the case of the majority of the scenarios where no special pattern to be detected, but only interested in the event alerts detection. In this case, our verifications are as follows,
```python
        if query_result['hits']['total']['value'] == 1 and \
``` 
- `whodata`: The case of `whodata` alerts.
- - In this case we apply some extra verification on the attributes we expect to exist in the alerts.
```python
        if query_result['hits']['total']['value'] == 1 and \
           query_result['hits']['hits'][0]['_source']['syscheck']\
               ['audit']['process']['name'] and \
           query_result['hits']['hits'][0]['_source']['syscheck']\
               ['audit']['process']['id'] and \
           query_result['hits']['hits'][0]['_source']['syscheck']\
               ['audit']['user']['name'] and \
           query_result['hits']['hits'][0]['_source']['syscheck']\
               ['audit']['user']['id']:
            success +=1
```

- `diff`: The case of `report_changes` alerts. We also search for some key attributes in the corresponding query result, as follows,
```python
        if query_result['hits']['total']['value'] == 1 and \
           query_result['hits']['hits'][0]['_source']['syscheck']\
               ['diff']['diff_statement']:
            success += 1
```

Regarding the `syscheck:tag`'s alerts verification, we initially add the tag terms as follows:
```
    if args.tag_query is not None:
        logging.info(
            "Setting tag query"
        )
        query["query"]["bool"]["filter"].append(
            {"terms": {"syscheck.tags": args.tag_query}}
        )
```

Where `args.tag_query` is the tag we are searching for.

---

We also changed the way we deal with the alerts. Now deal with alerts once they are loaded in a list object.

This way, and iterating over the list, each time we verify an alert successfully we pop it out of the list so we the list at the end of each retry, will contain only the alerts with failed verification; this way we reduce the execution time by no iterating over all the alerts again on each retry.

We now control the growth of the list of alerts in realtime, so after each retry and if the list still contains alerts, we check if the list of alerts recovered from ES index has grown in comparison to the last check or not, having a sleep time between each retry.

Kr,

Rshad
